client/src/components/ReviewTile.js

```diff
@@ -1,20 +1,107 @@
-import React from "react";
+import React, { useState, useEffect } from 'react'
+import { withRouter } from 'react-router-dom'
 
-const ReviewTile = ({ title, rating, body, recipe }) => {
-  let reviewDescription;
+const ReviewTile = ({ title, rating, body, recipe, user, id, voteCount, reviewersVoted }) => {
+
+  let reviewDescription
   if (body) {
     reviewDescription = <p>Review: {body}</p>;
   }
+
+  const [totalVotes, setTotalVotes] = useState(voteCount)
+  const [userVoted, setUserVoted] = useState(false)
+  
+  const getVotes = async () => {
+    try {
+      const response = await fetch("/api/v1/votes")
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
+      }
+      const responseBody = await response.json()
+      setTotalVotes(totalVotes, responseBody.value)
+    } catch (error) {
+      return console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  useEffect(() => {
+    getVotes()
+  }, [])
+
+  const addVote = async (value, userId) => {
+    try {
+      const response = await fetch(`/api/v1/votes/${id}/votes`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify({
+          value: value,
+          reviewId: id,
+          userId: userId
+        })
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+          alert(body.message)  
+        }
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
+      }
+      const responseBody = await response.json()
+      const newTotalVotes = parseInt(totalVotes) + responseBody.votes.value
+      setTotalVotes(newTotalVotes)
+      setUserVoted(true)
+    } catch (error) {
+      return console.error(`Error in fetch: ${error.message}`)
+    }
+  }
```

Comment:
once you refactor away the other votes get request, this will likely have to live higher up in your app (and thats okay. Thats what makes the showpage component a God component)